### DISCUSSION TITLE
How to check that key is revoked?

### DISCUSSION THREAD

**Discussion by IvanPizhenko** [openpgpjs/openpgpjs#1477] [Q&A] [answered]
Note: I'm using Typescript.
Warning: I'm quite new to openpgp.js and may ask questions that are obvious for more experienced people.
I'm trying to port an application that uses openpgp.js v4 to v5. 
Openpgp v4 have had `Key.isRevoked()` without parameters. In v5 it requires at least passing signature.
From `openpgp.d.ts`:
```typescript
  public isRevoked(signature: SignaturePacket, key?: AnyKeyPacket, date?: Date, config?: Config): Promise<boolean>;
```
What is equivalent call to v5 function?

**Comment by IvanPizhenko** [comment, marked as answer]
Looks like it still can be called with `signature === undefined`, but then why in `openpgp.d.ts` it is mandatory?
I've ended up with editing `openpgp.d.ts`.

**Comment by larabr** [comment]
Hi @IvanPizhenko , what's your use case? When trying to use the key, it internally checks whether the key is revoked.
Feel free to open a PR to fix the type definitions.

**Comment by IvanPizhenko** [comment]
@larabr Thanks for your answers, please find PR #1486

**Comment by IvanPizhenko** [comment]
@larabr The use case is here:
https://github.com/FlowCrypt/flowcrypt-ios/blob/e17f4d1b8cdb1e1f1c3823846a8e69b9cd3c88de/Core/source/core/pgp-key.ts#L370
